app/products/playbooks/screens/post_update/post_update.test.tsx

```diff
@@ -0,0 +1,732 @@
+            runName: 'Test Run',
+            userId: 'user-id',
+            channelId: 'channel-id',
+            teamId: 'team-id',
+            outstanding: 0,
+        };
+    }
+
+    const mockMetadata: PlaybookRunMetadata = {
+        followers: ['follower-1', 'follower-2'],
+        channel_name: 'channel-name',
+        channel_display_name: 'channel-display-name',
+        team_name: 'team-name',
+        num_participants: 10,
+        total_posts: 100,
+    };
+
+    const mockRun = TestHelper.fakePlaybookRun({
+        id: 'playbook-run-id',
+        status_update_broadcast_channels_enabled: true,
+        broadcast_channel_ids: ['channel-1', 'channel-2'],
+        reminder_message_template: 'Default message template',
+        status_posts: [],
+    });
+
+    const mockPost = TestHelper.fakePostModel({
+        id: 'post-id',
+        message: 'Last status post message',
+        createAt: Date.now(),
+    });
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+        jest.mocked(fetchPlaybookRunMetadata).mockResolvedValue({metadata: mockMetadata});
+        jest.mocked(fetchPlaybookRun).mockResolvedValue({run: mockRun});
+        jest.mocked(getPosts).mockResolvedValue([mockPost]);
+        jest.mocked(postStatusUpdate).mockResolvedValue(undefined);
+    });
+
+    it('should render loading state initially', () => {
+        const props = getBaseProps();
+        const {getByTestId} = renderWithIntlAndTheme(<PostUpdate {...props}/>);
+
+        expect(getByTestId('loader')).toBeTruthy();
+    });
+
+    it('should render correctly after loading', async () => {
+        const props = getBaseProps();
+        const {getByText, getByTestId} = renderWithIntlAndTheme(<PostUpdate {...props}/>);
+
+        await waitFor(() => {
+            expect(getByText(/This update for the run/)).toBeTruthy();
+            const input = getByTestId('FloatingTextInput');
+            expect(input).toHaveProp('value', 'Default message template');
+            expect(input).toHaveProp('label', 'Update message');
+            expect(input).toHaveProp('placeholder', 'Enter your update message');
+            expect(input).toHaveProp('multiline', true);
+
+            const selector = getByTestId('playbooks.post_update.selector');
+            expect(selector).toHaveProp('selected', '15_minutes');
+            expect(selector).toHaveProp('label', 'Timer for next update');
+            expect(selector).toHaveProp('options', expect.arrayContaining([
+                expect.objectContaining({value: '15_minutes', text: '15 minutes'}),
+                expect.objectContaining({value: '30_minutes', text: '30 minutes'}),
+                expect.objectContaining({value: '1_hour', text: '1 hour'}),
+                expect.objectContaining({value: '4_hours', text: '4 hours'}),
+                expect.objectContaining({value: '1_day', text: '1 day'}),
+                expect.objectContaining({value: '7_days', text: '7 days'}),
+            ]));
```

Comment:
since you are checking one by one, maybe throw a length comparison so no other items are there.